# <h> Exploratory Data Analysis of Range, Efficiency, and Price of Electric Vehicles</h>

This project utilizes a data set on brand, range, efficiency, body style, and price for 103 electric vehicles from 33 top car brands. There are four parts to this analysis:

1. Data are read in and cleaned for missing values and duplicates.
2. The top brands are identified, and descriptive statistics and data visualizations are displayed for range, efficiency, and price by brand. 
3. Descriptive statistics and data visulizations are run to show the range, efficiency, and price by body style. 
4. Finally, the relationship of price vs range and price vs efficiency are displayed in scatterplots. 

## <b> 1. Import and Clean Data

In this section, the Python libraries and data set are imported into Jupyter Notebook. The data are checked for duplicates, missing values, and any other potential issues.

### 1a. Import Libraries and Data

In [2]:
# Import Libraries
import pandas as pd 
import numpy as np
from matplotlib import pyplot as plt
import plotly.express as px

In [3]:
# Read in dataset
ev = pd.read_csv('/Users/kellyshreeve/Desktop/Data-Sets/ElectricCarData_Clean.csv')

In [4]:
# Print dataset info
ev.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 103 entries, 0 to 102
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Brand            103 non-null    object 
 1   Model            103 non-null    object 
 2   AccelSec         103 non-null    float64
 3   TopSpeed_KmH     103 non-null    int64  
 4   Range_Km         103 non-null    int64  
 5   Efficiency_WhKm  103 non-null    int64  
 6   FastCharge_KmH   103 non-null    object 
 7   RapidCharge      103 non-null    object 
 8   PowerTrain       103 non-null    object 
 9   PlugType         103 non-null    object 
 10  BodyStyle        103 non-null    object 
 11  Segment          103 non-null    object 
 12  Seats            103 non-null    int64  
 13  PriceEuro        103 non-null    int64  
dtypes: float64(1), int64(5), object(8)
memory usage: 11.4+ KB


There are 103 entries in this data set. FastCharge_KmH was read in as object and needs to be changed to int. Variable names need to be changed to snake case. There do not appear to be any missing values in this data set.

In [5]:
# Print the first 10 rows of the dataset
ev.head(10)

Brand                          Model  AccelSec  TopSpeed_KmH  \
0       Tesla   Model 3 Long Range Dual Motor       4.6           233   
1  Volkswagen                       ID.3 Pure      10.0           160   
2    Polestar                               2       4.7           210   
3         BMW                            iX3        6.8           180   
4       Honda                              e        9.5           145   
5       Lucid                            Air        2.8           250   
6  Volkswagen                         e-Golf        9.6           150   
7     Peugeot                          e-208        8.1           150   
8       Tesla     Model 3 Standard Range Plus       5.6           225   
9        Audi                      Q4 e-tron        6.3           180   

   Range_Km  Efficiency_WhKm FastCharge_KmH RapidCharge PowerTrain  \
0       450              161            940         Yes        AWD   
1       270              167            250         Yes        RWD   
2       400              181            620         Yes        AWD   
3       360              206            560         Yes        RWD   
4       170              168            190         Yes        RWD   
5       610              180            620         Yes        AWD   
6       190              168            220         Yes        FWD   
7       275              164            420         Yes        FWD   
8       310              153            650         Yes        RWD   
9       400              193            540         Yes        AWD   

     PlugType  BodyStyle Segment  Seats  PriceEuro  
0  Type 2 CCS      Sedan       D      5      55480  
1  Type 2 CCS  Hatchback       C      5      30000  
2  Type 2 CCS   Liftback       D      5      56440  
3  Type 2 CCS        SUV       D      5      68040  
4  Type 2 CCS  Hatchback       B      4      32997  
5  Type 2 CCS      Sedan       F      5     105000  
6  Type 2 CCS  Hatchback       C      5      31900  
7  Type 2 CCS  Hatchback       B      5      29682  
8  Type 2 CCS      Sedan       D      5      46380  
9  Type 2 CCS        SUV       D      5      55000

### 1b. Rename Columns and Change Data Types

In [6]:
# Rename columns with snake case

ev = ev.rename(
    columns={'Brand':'brand',
             'Model':'model',
             'AccelSec':'accel_sec',
             'TopSpeed_KmH':'top_speed_kmh',
             'Range_Km':'range_km',
             'Efficiency_WhKm':'efficiency_whkm',
             'FastCharge_KmH':'fast_charge_kmh',
             'RapidCharge':'rapid_charge',
             'PowerTrain':'power_train',
             'PlugType':'plug_type',
             'BodyStyle':'body_style',
             'Segment':'segment',
             'Seats':'seats',
             'PriceEuro':'price_euro'
             }
)

In [7]:
# Check columns are now all snake case
print(ev.columns)

Index(['brand', 'model', 'accel_sec', 'top_speed_kmh', 'range_km',
       'efficiency_whkm', 'fast_charge_kmh', 'rapid_charge', 'power_train',
       'plug_type', 'body_style', 'segment', 'seats', 'price_euro'],
      dtype='object')


In [8]:
# Change fast_charge_kmh from object to int

# First check the unique values of fast_charge_kmh
print(ev['fast_charge_kmh'].unique())

['940' '250' '620' '560' '190' '220' '420' '650' '540' '440' '230' '380'
 '210' '590' '780' '170' '260' '930' '850' '910' '490' '470' '270' '450'
 '350' '710' '240' '390' '570' '610' '340' '730' '920' '-' '550' '900'
 '520' '430' '890' '410' '770' '460' '360' '810' '480' '290' '330' '740'
 '510' '320' '500']


In [9]:
ev['fast_charge_kmh'] = ev['fast_charge_kmh'].replace('-', np.NaN) # Change '-' to NaN

print(ev['fast_charge_kmh'].unique())

['940' '250' '620' '560' '190' '220' '420' '650' '540' '440' '230' '380'
 '210' '590' '780' '170' '260' '930' '850' '910' '490' '470' '270' '450'
 '350' '710' '240' '390' '570' '610' '340' '730' '920' nan '550' '900'
 '520' '430' '890' '410' '770' '460' '360' '810' '480' '290' '330' '740'
 '510' '320' '500']


In [10]:
ev['fast_charge_kmh'] = pd.to_numeric(ev['fast_charge_kmh']) # Change fast_charge_kmh to int type data

ev.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 103 entries, 0 to 102
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   brand            103 non-null    object 
 1   model            103 non-null    object 
 2   accel_sec        103 non-null    float64
 3   top_speed_kmh    103 non-null    int64  
 4   range_km         103 non-null    int64  
 5   efficiency_whkm  103 non-null    int64  
 6   fast_charge_kmh  98 non-null     float64
 7   rapid_charge     103 non-null    object 
 8   power_train      103 non-null    object 
 9   plug_type        103 non-null    object 
 10  body_style       103 non-null    object 
 11  segment          103 non-null    object 
 12  seats            103 non-null    int64  
 13  price_euro       103 non-null    int64  
dtypes: float64(2), int64(5), object(7)
memory usage: 11.4+ KB


### 1c. Check for Missing Data and Duplicates

In [11]:
# Check for missing values
ev.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 103 entries, 0 to 102
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   brand            103 non-null    object 
 1   model            103 non-null    object 
 2   accel_sec        103 non-null    float64
 3   top_speed_kmh    103 non-null    int64  
 4   range_km         103 non-null    int64  
 5   efficiency_whkm  103 non-null    int64  
 6   fast_charge_kmh  98 non-null     float64
 7   rapid_charge     103 non-null    object 
 8   power_train      103 non-null    object 
 9   plug_type        103 non-null    object 
 10  body_style       103 non-null    object 
 11  segment          103 non-null    object 
 12  seats            103 non-null    int64  
 13  price_euro       103 non-null    int64  
dtypes: float64(2), int64(5), object(7)
memory usage: 11.4+ KB


After changing '-' to NaN, there do appear to be missing values in fast_charge_kmh. These represent cars that don't have measurements for this value. Because there are only 5 missing values, I will keep the missing data and exclude these cars from analyses that use this variable. 

In [12]:
# Check for fully duplicate rows
ev_duplicates = ev.duplicated().sum()

print(f'The number of fully duplicated rows is: {ev_duplicates}')

The number of fully duplicated rows is: 0


In [13]:
ev_brand_model_duplicates = ev[['brand', 'model']].duplicated().sum() # Check for implicit brand-model duplicate

print(f'The number of brand-model duplicates is: {ev_brand_model_duplicates}')

The number of brand-model duplicates is: 1


In [14]:
ev_brand_model_duplicate_row = ev[ev[['brand', 'model']].duplicated()==True] # Find the duplicate brand-model rows

display(ev_brand_model_duplicate_row) # Display the duplicate row

display(ev[(ev['brand']=='Kia ') & (ev['model']=='e-Soul 64 kWh')]) # Display the original and duplicated rows

brand          model  accel_sec  top_speed_kmh  range_km  efficiency_whkm  \
92  Kia   e-Soul 64 kWh        7.9            167       365              175   

    fast_charge_kmh rapid_charge power_train   plug_type body_style segment  \
92            320.0          Yes         FWD  Type 2 CCS        SUV       B   

    seats  price_euro  
92      5       36837

brand          model  accel_sec  top_speed_kmh  range_km  efficiency_whkm  \
45  Kia   e-Soul 64 kWh        7.9            167       365              175   
92  Kia   e-Soul 64 kWh        7.9            167       365              175   

    fast_charge_kmh rapid_charge power_train   plug_type body_style segment  \
45            340.0          Yes         FWD  Type 2 CCS        SUV       B   
92            320.0          Yes         FWD  Type 2 CCS        SUV       B   

    seats  price_euro  
45      5       36837  
92      5       36837

The Kia e-Soul 64 kWh row is fully duplicated except for the fast_charge_kmh column. I believe this was a data entry error, that someone mis-entered either the 340 or 320. I believe that there shold be only one make and model of this car. Therefore I will delete this duplicate row. 

In [15]:
ev = ev.drop_duplicates(subset=['brand', 'model']).reset_index(drop=True) # Delete duplicate rows

new_ev_brand_model_duplicates = ev[['brand', 'model']].duplicated().sum() # Check again for duplicates

print(f'The number of duplicates in the ev dataset is now: {new_ev_brand_model_duplicates}')

The number of duplicates in the ev dataset is now: 0


## 2. Exploratory Data Analysis

This section details descriptive statistics and data visualization for range, efficiency, and price of electric vehicles by brand and by body style.

Additionally, the relationship of range vs price and efficiency vs price is visualized on a scatter plot.

### 2a. Create a top_brands data frame and run descriptive analysis of price, efficiency, and range by brand

I define top_brand as brands that have 3 or more electric vehicles on the market.

In [16]:
n_brand_unique = len(ev['brand'].unique()) # Count unique brands represented in the dataset

print(f'There are {n_brand_unique} unique car brands. They are:')
print()
print(ev['brand'].unique()) # Display unique brand names

There are 33 unique car brands. They are:

['Tesla ' 'Volkswagen ' 'Polestar ' 'BMW ' 'Honda ' 'Lucid ' 'Peugeot '
 'Audi ' 'Mercedes ' 'Nissan ' 'Hyundai ' 'Porsche ' 'MG ' 'Mini ' 'Opel '
 'Skoda ' 'Volvo ' 'Kia ' 'Renault ' 'Mazda ' 'Lexus ' 'CUPRA ' 'SEAT '
 'Lightyear ' 'Aiways ' 'DS ' 'Citroen ' 'Jaguar ' 'Ford ' 'Byton '
 'Sono ' 'Smart ' 'Fiat ']


In [17]:
ev['brand'].value_counts() # Count the number of vehiclese for each brand in the dataset

Tesla          13
Audi            9
Nissan          8
Volkswagen      8
Skoda           6
Renault         5
Porsche         5
BMW             4
Ford            4
Kia             4
Smart           3
Byton           3
Mercedes        3
Hyundai         3
Opel            3
Fiat            2
Peugeot         2
Honda           2
Mini            1
DS              1
Polestar        1
Sono            1
Lucid           1
Jaguar          1
Citroen         1
Aiways          1
MG              1
Lightyear       1
SEAT            1
CUPRA           1
Lexus           1
Mazda           1
Volvo           1
Name: brand, dtype: int64

In [18]:
# Create data frame with only brands that have 3 or more evs on the market
top_brands = ev[(ev['brand']=='Tesla ') | (ev['brand']=='Audi ') | (ev['brand']=='Nissan ') 
                | (ev['brand']=='Volkswagen ') | (ev['brand']=='Skoda ') | (ev['brand']=='Renault ') 
                | (ev['brand']=='Porsche ') | (ev['brand']=='BMW ') | (ev['brand']=='Ford ')
                | (ev['brand']=='Kia ') | (ev['brand']=='Smart ') | (ev['brand']=='Byton ')
                | (ev['brand']=='Mercedes ') | (ev['brand']=='Hyundai ') | (ev['brand']=='Opel ')] 

In [31]:
# Create a pivot table with min, max, mean, median, and standard deviation for each brand's
# efficiency, price, and range.
top_brands_pivot = top_brands.pivot_table(index='brand', values=['range_km', 'efficiency_whkm', 'price_euro'], 
                                          aggfunc=['min', 'max', 'mean', 'median', 'std']).round(2)

display(top_brands_pivot)

min                                 max             \
            efficiency_whkm price_euro range_km efficiency_whkm price_euro   
brand                                                                        
Audi                    188      55000      280             270     125000   
BMW                     161      38017      230             206      68040   
Byton                   222      53500      325             244      64000   
Ford                    194      46900      340             209      62900   
Hyundai                 153      33971      250             160      40795   
Kia                     167      33133      230             175      38105   
Mercedes                171      45000      330             273      70631   
Nissan                  164      29234      190             232      65000   
Opel                    164      29146      255             176      41906   
Porsche                 195     102945      365             223     180781   
Renault                 161      24790      130             194      38000   
Skoda                   166      24534      195             193      47500   
Smart                   167      21387       95             176      24565   
Tesla                   153      45000      310             267     215000   
Volkswagen              166      21421      190             183      45000   

                                mean                              median  \
            range_km efficiency_whkm price_euro range_km efficiency_whkm   
brand                                                                      
Audi             425          224.56   80593.67   356.67           228.0   
BMW              450          177.50   53145.75   318.75           171.5   
Byton            400          234.67   59833.33   371.67           238.0   
Ford             450          202.25   54568.75   395.00           203.0   
Hyundai          400          155.67   36408.33   301.67           154.0   
Kia              370          171.25   35618.75   300.00           171.5   
Mercedes         370          220.00   61705.00   350.00           216.0   
Nissan           440          194.75   45902.12   328.12           196.0   
Opel             335          171.00   35350.67   288.33           173.0   
Porsche          425          209.40  138265.80   388.00           215.0   
Renault          315          170.40   31268.20   234.00           165.0   
Skoda            420          182.50   38255.67   337.50           182.0   
Smart            100          173.00   22660.67    96.67           176.0   
Tesla            970          201.38   80272.31   500.77           188.0   
Volkswagen       440          170.88   34602.38   318.12           169.5   

                                            std                      
            price_euro range_km efficiency_whkm price_euro range_km  
brand                                                                
Audi           79445.0    365.0           28.54   22003.17    51.84  
BMW            53263.0    297.5           20.34   15559.13   106.17  
Byton          62000.0    390.0           11.37    5575.24    40.72  
Ford           54237.5    395.0            6.65    6546.03    53.23  
Hyundai        34459.0    255.0            3.79    3806.79    85.20  
Kia            35618.5    300.0            3.50    2260.52    78.00  
Mercedes       69484.0    350.0           51.12   14478.32    20.00  
Nissan         47500.0    327.5           20.89   12228.34    87.79  
Opel           35000.0    275.0            6.24    6387.22    41.63  
Porsche       148301.0    385.0           12.60   32141.87    22.80  
Renault        31184.0    255.0           13.43    4871.00    85.25  
Skoda          38750.0    360.0           10.07    8166.28    86.59  
Smart          22030.0     95.0            5.20    1680.25     2.89  
Tesla          65620.0    450.0           39.08   44432.09   173.20  
Volkswagen     34287.5    340.0            5.79    7301.64  

In [172]:
# Create a bar chart of average price by brand
price_bar = px.histogram(top_brands, x='brand', y='price_euro', histfunc='avg', 
                          title='Average Price by Brand of EVs', text_auto='.2s',
                          labels={'price_euro':'Price (Euros)', 'brand':'Brand Name'},
                          color_discrete_sequence=[px.colors.qualitative.Plotly[7]],
                          width=800, height=500)

price_bar.update_layout({
    'plot_bgcolor':'rgba(0, 0, 0, 0)',
    'paper_bgcolor':'rgba(0, 0, 0, 0)'
}) # Turn off background color

price_bar.update_traces(textfont_size=11, textposition='outside') # Add labels above bars

price_bar.update_layout(xaxis={'categoryorder':'total descending'}) # Arrange in order from expensive to inexpensive

price_bar.update_xaxes(showgrid=False) # Turn off x grid
price_bar.update_yaxes(showgrid=False) # Turn off y grid

price_bar.show()

In [173]:
# Create a bar chart of average efficiency by brand
eff_bar = px.histogram(top_brands, x='brand', y='efficiency_whkm', histfunc='avg', 
                          title='Average Efficiency by Brand of EVs', text_auto='.2s',
                          labels={'efficiency_whkm':'Efficiency (WhKm)', 'brand':'Brand Name'},
                          color_discrete_sequence=[px.colors.qualitative.Plotly[7]],
                          width=800, height=500)

eff_bar.update_layout({
    'plot_bgcolor':'rgba(0, 0, 0, 0)',
    'paper_bgcolor':'rgba(0, 0, 0, 0)'
}) 

eff_bar.update_traces(textfont_size=11, textposition='outside')

# Arrange the bars in the same order as the price chart
eff_bar.update_layout(xaxis={'categoryorder':'array', 'categoryarray':
    ['Porsche ', 'Audi ', 'Tesla ', 'Mercedes ', 'Byton ', 'Ford ', 'BMW ',
     'Nissan ', 'Skoda ', 'Hyundai ', 'Kia ', 'Opel ', 'Volkswagen ', 'Renault ', 'Smart ']})

eff_bar.update_xaxes(showgrid=False)
eff_bar.update_yaxes(range=[0,250], showgrid=False) # Set y axis range

eff_bar.show()

In [54]:
fig = px.colors.qualitative.swatches()
fig.show()

In [170]:
# Create a bar chart of average range by brand
range_bar = px.histogram(top_brands, x='brand', y='range_km', histfunc='avg', 
                          title='Average Range by Brand of EVs', text_auto='.2s',
                          labels={'range_km':'Range (Km)', 'brand':'Brand Name'},
                          color_discrete_sequence=[px.colors.qualitative.Plotly[7]],
                          width=800, height=500)

range_bar.update_layout({
    'plot_bgcolor':'rgba(0, 0, 0, 0)',
    'paper_bgcolor':'rgba(0, 0, 0, 0)'
})

range_bar.update_traces(textfont_size=11, textposition='outside')

range_bar.update_layout(xaxis={'categoryorder':'array', 'categoryarray':
    ['Porsche ', 'Audi ', 'Tesla ', 'Mercedes ', 'Byton ', 'Ford ', 'BMW ',
     'Nissan ', 'Skoda ', 'Hyundai ', 'Kia ', 'Opel ', 'Volkswagen ', 'Renault ', 'Smart ']})

range_bar.update_xaxes(showgrid=False)
range_bar.update_yaxes(showgrid=False)

range_bar.show()

### 2b. Run descriptive analysis for efficiency, range, and price by body style

In [23]:
ev['body_style'].value_counts() # Find how many of each body style are in the data frame

SUV          44
Hatchback    32
Sedan        10
Liftback      5
Pickup        3
Cabrio        3
SPV           3
MPV           1
Station       1
Name: body_style, dtype: int64

In [24]:
# Create data frame with only top 3 body styles
suv_hatch_sed = ev[(ev['body_style']=='SUV') | (ev['body_style']=='Hatchback') | (ev['body_style']=='Sedan')]

In [25]:
# Run pivot table for min, max, mean, median, and standard deviation for each body style
suv_hatch_sed_pivot = suv_hatch_sed.pivot_table(index='body_style', values=['range_km', 'efficiency_whkm', 'price_euro'], 
                                          aggfunc=['min', 'max', 'mean', 'median', 'std'])

display(suv_hatch_sed_pivot)

min                                 max             \
           efficiency_whkm price_euro range_km efficiency_whkm price_euro   
body_style                                                                  
Hatchback              156      20129       95             232      65000   
SUV                    154      30000      180             270     102990   
Sedan                  153      46380      310             223     180781   

                               mean                                    median  \
           range_km efficiency_whkm    price_euro    range_km efficiency_whkm   
body_style                                                                      
Hatchback       440      172.687500  34893.156250  268.593750           167.5   
SUV             450      198.090909  54372.204545  344.090909           193.0   
Sedan           610      186.600000  99966.900000  423.500000           187.5   

                                           std                           
           price_euro range_km efficiency_whkm    price_euro   range_km  
body_style                                                               
Hatchback     32998.5    262.5       15.883523  10516.855964  95.446610  
SUV           51750.0    362.5       26.733672  19073.516759  70.754018  
Sedan        103972.5    425.0       22.862876  43677.697125  78.953221

In [169]:
# Create a histogram of price by body style
price_hist = px.histogram(suv_hatch_sed, title='Price by Body Style for EVs', x='price_euro', color='body_style', nbins=30,
                          labels={'price_euro':'Price (Euros)', 'count':'Frequency'},
                          color_discrete_sequence=[px.colors.qualitative.Plotly[0],
                                                   px.colors.qualitative.Plotly[7],
                                                   px.colors.qualitative.Plotly[9]],
                          width=800, height=500)
                                                   

price_hist.update_layout({
    'plot_bgcolor':'rgba(0, 0, 0, 0)',
    'paper_bgcolor':'rgba(0, 0, 0, 0)'
})

price_hist.update_layout(barmode='overlay')
price_hist.update_traces(opacity=0.7)

price_hist.update_xaxes(showgrid=False)
price_hist.update_yaxes(showgrid=False)

price_hist.show()

In [168]:
# Create a histogram of efficiency by body style
eff_hist = px.histogram(suv_hatch_sed, title='Efficiency by Body Style for EVs', x='efficiency_whkm', color='body_style', nbins=22,
                          labels={'efficiency_whkm':'Efficiency (WhKm)', 'count':'Frequency'},
                          color_discrete_sequence=[px.colors.qualitative.Plotly[0],
                                                   px.colors.qualitative.Plotly[7],
                                                   px.colors.qualitative.Plotly[9]],
                          width=800, height=500)
                                                   

eff_hist.update_layout({
    'plot_bgcolor':'rgba(0, 0, 0, 0)',
    'paper_bgcolor':'rgba(0, 0, 0, 0)'
})

eff_hist.update_layout(barmode='overlay')
eff_hist.update_traces(opacity=0.7)

eff_hist.update_xaxes(showgrid=False)
eff_hist.update_yaxes(showgrid=False)

eff_hist.show()

In [167]:
# Create a histogram of range by body style
range_hist = px.histogram(suv_hatch_sed, title='Range by Body Style for EVs', x='range_km', color='body_style', nbins=25,
                          labels={'range_km':'Range (Km)', 'count':'Frequency'},
                          color_discrete_sequence=[px.colors.qualitative.Plotly[0],
                                                   px.colors.qualitative.Plotly[7],
                                                   px.colors.qualitative.Plotly[9]],
                          width=800, height=500)
                                                   

range_hist.update_layout({
    'plot_bgcolor':'rgba(0, 0, 0, 0)',
    'paper_bgcolor':'rgba(0, 0, 0, 0)'
})

range_hist.update_layout(barmode='overlay')
range_hist.update_traces(opacity=0.7)

range_hist.update_xaxes(showgrid=False)
range_hist.update_yaxes(showgrid=False)

range_hist.show()

### 2c. Scatter plots of efficiency by price and range by price

In [185]:
price_efficiency = px.scatter(data_frame=ev, title='Efficiency vs Price for EVs', x='price_euro', y='efficiency_whkm', 
           labels={'price_euro':'Price Euro', 'efficiency_whkm':'Efficiency WhKm'},
           color='brand', width=900, height=500)

price_efficiency.update_layout({
    'plot_bgcolor':'rgba(0, 0, 0, 0)',
    'paper_bgcolor':'rgba(0, 0, 0, 0)'
})

price_efficiency.update_xaxes(range=[0, 250000], showgrid=False)
price_efficiency.update_yaxes(range=[125, 280], showgrid=False)

In [187]:
price_range = px.scatter(data_frame=ev, title='Range vs Price for EVs', x='price_euro', y='range_km', 
           labels={'price_euro':'Price Euro', 'range_km':'Range Km'},
           color='brand', width=900, height=500)

price_range.update_layout({
    'plot_bgcolor':'rgba(0, 0, 0, 0)',
    'paper_bgcolor':'rgba(0, 0, 0, 0)'
})

price_range.update_xaxes(range=[0, 250000], showgrid=False)
price_range.update_yaxes(showgrid=False)
